crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3349,26 +3349,58 @@ impl<'db> TypeInferenceBuilder<'db> {
 
             // Walk up parent scopes looking for a possible enclosing scope that may have a
             // definition of this name visible to us (would be `LOAD_DEREF` at runtime.)
-            for (enclosing_scope_file_id, _) in self.index.ancestor_scopes(file_scope_id) {
+            // Note that we skip the scope containing the use that we are resolving, since we
+            // already looked for the symbol there up above.
+            let mut look_up_eagerly = scope.is_eager(db);
+            for (enclosing_scope_file_id, _) in self.index.ancestor_scopes(file_scope_id).skip(1) {
```

Comment:
Nice catch! Shouldn't ever cause a correctness problem, but definitely meant we were doing unnecessary work.